### PR TITLE
[BUG] Sendinblue Transactional Email action fails when trying to add recipients to **To** field

### LINKED ISSUE #3514: [BUG] Sendinblue Transactional Email action fails when trying to add recipients to **To** field
**Describe the bug**
Sendinblue Transactional Email action fails when trying to manually add recipients to **To** field

**To Reproduce**
1. Create a new workflow, add SendInBlue: Send Transactional Email
3. On the `To` prop, **manually** add the JSON data using Custom Expression. For example: `{{ [{"email":"test@example","name":"Test User"}] }}`
4. Observe the error
```
SyntaxError
Unexpected token o in JSON at position 1
```
FireShot Capture 396 - Pipedream - Connect APIs, Remarkably Fast - pipedream com

**Expected behavior**
1. On the `To` prop, the User can manually add the JSON data using Custom Expression. 
2. The `To` prop need to be able to accept JSON Object (.i.e `{"email":"test@example","name":"Test User"}`) and JSON Array (.i.e `[{"email":"test@example","name":"Test User"}]`)

**Additional context**
[Reported by a customer here](https://pipedream-users.slack.com/archives/CMZG4EBJ9/p1658670184747329)

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #3514 

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/3679"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/sendinblue/actions/send-transactional-email/send-transactional-email.mjs [hunk 1]
```diff
@@ -108,6 +108,14 @@ export default {
 
     return props;
   },
+  methods: {
+    formatEmailProp(prop) {
+      if (typeof prop[0] === "string") {
+        return Object.keys(prop).map((key) => JSON.parse(prop[key]));
+      }
+      return prop;
+    },
```
Comment (jcortes) [inline components/sendinblue/actions/send-transactional-email/send-transactional-email.mjs:117]: Optional if you want to improve a bit more
```suggestion
function formatEmailProp(prop) {
  if (!Array.isArray(prop)) {
    throw new Error("Field should be an array of strings", prop);
  }
  return Object.values(prop)
    .reduce((reduction, value) => {
      try {
        if (typeof(value) !== "string") {
          throw "Not the right type";
        }
        return reduction.concat(JSON.parse(value));
      } catch (error) {
        console.log(`Error parsing value "${value}"`);
        return reduction;
      }
    }, []);
}
```

Or this version would be much simpler
```suggestion
function formatEmailProp(prop) {
  if (!Array.isArray(prop)) {
    throw new Error("Field should be an array of strings", prop);
  }
  return Object.values(prop).map((value) => JSON.parse(value));
}
```

**Review by jcortes — COMMENTED** [review]
Hi @michelle0927 Just added one optional observation, I will leave it to you if you want to change it either way I think your version works. thanks! Ready for QA

File: components/sendinblue/actions/send-transactional-email/send-transactional-email.mjs (hunk 1 above)
Comment (michelle0927) [reply components/sendinblue/actions/send-transactional-email/send-transactional-email.mjs]: Thanks. I updated it to check that `prop` is an array and to throw an error if not. The issue for this PR is requesting that the fields allow an array of objects, so I don't want to force them to be an array of strings. We just don't want to try and parse the values if they're already objects.

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Sendinblue_3679.pdf](https://github.com/PipedreamHQ/pipedream/files/9200444/Sendinblue_3679.pdf)

**Review by jcortes — APPROVED** [review]
Approved!

**Pull request merged**